addon/components/frost-component.js

```diff
@@ -31,19 +31,23 @@ export default Component.extend(PropTypesMixin, HookMixin, SpreadMixin, CssMixin
       PropTypes.null
     ]),
     attributeBindings: PropTypes.arrayOf(PropTypes.string),
-    classNameBindings: PropTypes.arrayOf(PropTypes.string),
-    classNames: PropTypes.arrayOf(PropTypes.string),
+    // FIXME: the following property defined in propTypes cause validation errors
+    // when ember-prop-types validateOnUpdate setting is set to true
+    // classNameBindings: PropTypes.arrayOf(PropTypes.string),
+    // classNames: PropTypes.arrayOf(PropTypes.string),
     concatenatedProperties: PropTypes.arrayOf(PropTypes.string),
-    element: PropTypes.oneOfType([
-      PropTypes.element,
-      PropTypes.null
-    ]),
-    elementId: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.null
-    ]),
-    isDestroyed: PropTypes.bool,
-    isDestroying: PropTypes.bool,
+    // FIXME: the following properties defined in propTypes cause validation errors
+    // when ember-prop-types validateOnUpdate setting is set to true
+    // element: PropTypes.oneOfType([
+    //   PropTypes.element,
+    //   PropTypes.null
+    // ]),
+    // elementId: PropTypes.oneOfType([
+    //   PropTypes.string,
+    //   PropTypes.null
+    // ]),
+    // isDestroyed: PropTypes.bool,
+    // isDestroying: PropTypes.bool,
```

Comment:
I think this stuff really will fall under a way to mark properties as private with `ember-prop-types`, at which point we probably just won't check private properties on updates to mitigate this issue.